holmes/core/tool_calling_llm.py

```
@@ -565,11 +565,44 @@ def _get_tools(self) -> list:
+        Supported by litellm for all Anthropic-compatible providers
+        (direct API, Bedrock, Vertex AI). BM25 variant is not used
+        because it is not supported on Bedrock.
+
+        See: https://docs.litellm.ai/docs/providers/anthropic_tool_search
+        """
+        tool_search_tool = {
+            "type": "tool_search_tool_regex_20251119",
+            "name": "tool_search_tool_regex",
+        }
+        deferred_tools = []
+        for tool in tools:
```

Ruff (0.15.18)

[error] Import `tool` from line 3 shadowed by loop variable

(F402)